### Import Libraries

In [ ]:
import pandas as pd
from pathlib import Path

### Config

In [2]:
# Constants
RANDOM_STATE = 42
TARGET_COL = "label"
TEXT_COL = "text"

# Base path
BASE_DIR = Path.cwd().resolve().parents[1]

# Paths
DATA_DIR = BASE_DIR / "data"
PROCESSED_DIR = DATA_DIR / "processed"
PROCESSED_DATA_PATH = PROCESSED_DIR / "cleaned_sms.csv"

### Load Dataset

In [ ]:
df = pd.read_csv(PROCESSED_DATA_PATH)
print("Shape:", df.shape)
print("\nData Types:\n", df.dtypes)

Shape: (5157, 3)

Data Types:
 label       object
raw_text    object
text        object
dtype: object


In [4]:
# Preview
df.head(5)

,label,raw_text,text
0,ham,"Go until jurong point, crazy.. Available only ...","Go until jurong point, crazy.. Available only ..."
1,ham,Ok lar... Joking wif u oni...,Ok lar... Joking wif you oni...
2,spam,Free entry in 2 a wkly comp to win FA Cup fina...,Free entry in 2 a wkly comp to win FA Cup fina...
3,ham,U dun say so early hor... U c already then say...,YOU dun say so early hor... YOU c already then...
4,ham,"Nah I don't think he goes to usf, he lives aro...","Nah I do not think he goes to usf, he lives ar..."


In [ ]:
# Define Features and Target
X = df[TEXT_COL]
y = df[TARGET_COL]

In [ ]:
# Class distribution
print(y.value_counts(normalize=True).rename("ratio"))

label
ham     0.875703
spam    0.124297
Name: ratio, dtype: float64


**Observation**
- Processed dataset loaded successfully with 5157 rows and 3 columns
- `text` column will be used as input features, and `label` as target
- Class distribution remains imbalanced (~87.6% ham), consistent with preprocessing stage